odata-parser.js

```diff
@@ -1237,28 +1237,25 @@
         },
         Boolean: function() {
             var $elf = this, _fromIdx = this.input.idx;
-            return this._or(function() {
-                return this._apply("True");
-            }, function() {
-                return this._apply("False");
-            });
-        },
-        True: function() {
-            var $elf = this, _fromIdx = this.input.idx;
-            this._applyWithArgs("exactly", "t");
-            this._applyWithArgs("exactly", "r");
-            this._applyWithArgs("exactly", "u");
-            this._applyWithArgs("exactly", "e");
-            return !0;
-        },
-        False: function() {
-            var $elf = this, _fromIdx = this.input.idx;
-            this._applyWithArgs("exactly", "f");
-            this._applyWithArgs("exactly", "a");
-            this._applyWithArgs("exactly", "l");
-            this._applyWithArgs("exactly", "s");
-            this._applyWithArgs("exactly", "e");
-            return !1;
+            return function() {
+                switch (this.anything()) {
+                  case "f":
+                    this._applyWithArgs("exactly", "a");
+                    this._applyWithArgs("exactly", "l");
+                    this._applyWithArgs("exactly", "s");
+                    this._applyWithArgs("exactly", "e");
```

Comment:
It would be great if the parser could look ahead a certain number of characters and compare the result, this could then compare directly to “false” or “true” and would probably be a bit faster (assuming applyWithArgs’s implementation is not trivial)

Reply:
I expect it could well be slower because it would have to do the work to read the next n parts into a string and then do the comparison rather than doing it as it goes along and bailing early.  Bearing in mind each "part" could be a length > 1 string or an array or an object or whatever, there's no guarantee that it has to be a single character (and in a lot of our other grammars being a single character is the exception as they're working on ASTs instead)